File: sdk/javascript/examples/codex-agent.mjs
import { SynapseClient, createAgentDefinition } from "../src/index.js";

const agentId = "codex-js-example";
const client = new SynapseClient({
  baseUrl: "http://127.0.0.1:8000",
  agentId
});

await client.registerAgent(
  createAgentDefinition({
    agentId,
    kind: "codex",
    name: "Codex JS Example",
    description: "Example Codex-style agent using the JavaScript SDK.",
    allowedDomains: ["example.com"],
    allowedTools: ["github.search"]
  })
);

const browser = client.browser;
const page = await browser.open("https://example.com");
await browser.click("body");
const repos = await browser.callTool("github.search", {
  query: "browser agents python",
  per_page: 3
});

console.log({
  title: page.page.title,
  repos
});

File: examples/codex_agent.py
from synapse.models.agent import AgentDefinition, AgentKind
from synapse.sdk import SynapseClient


def main() -> None:
    client = SynapseClient()
    browser = client.browser

    client.register_agent(
        AgentDefinition(
            agent_id="codex-example",
            kind=AgentKind.CODEX,
            name="Codex Example",
            description="Example Codex Synapse SDK agent.",
        )
    )

    page = browser.open("https://example.com")
    heading = browser.extract("h1")
    repo_search = browser.call_tool("github.search", {"query": "browser agents python", "per_page": 3})
    browser.send_agent_message(
        sender_agent_id="codex-example",
        recipient_agent_id="claude-code-example",
        content="Codex inspected the landing page and GitHub ecosystem.",
        metadata={"links": page.page.links[:3]},
    )

    print({"heading": heading.model_dump(), "repos": repo_search})
    client.close()


if __name__ == "__main__":
    main()
